supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/api/ActionBuilder.scala

```diff
@@ -81,4 +94,192 @@ object ActionBuilder {
       .withFlags(flags)
       .withMounts(mounts)
       .withLabels(labels)
+
+  /**
+    * Returns a set of labels for a parameter.
+    *
+    * @param pipelinesParameter Input or output parameter to label.
+    * @return The labels.
+    */
+  def parameterLabels(pipelinesParameter: PipelinesParameter): Map[String, String] = {
+    pipelinesParameter match {
+      case _: PipelinesApiInput =>
+        Map(
+          Key.Tag -> Value.Localization,
+          Key.InputName -> pipelinesParameter.name
+        )
+      case _: PipelinesApiOutput =>
+        Map(
+          Key.Tag -> Value.Delocalization,
+          Key.OutputName -> pipelinesParameter.name
+        )
+    }
+  }
+
+  /**
+    * Surrounds the list of Actions with a pair of starting and done Actions.
+    *
+    * @param description       Description of the list of Actions.
+    * @param loggingLabelValue An entry from Value that describes the list of Actions.
+    * @param isAlwaysRun       If true the pair of starting and done Actions will always be run.
+    * @param actions           The list of Actions to surround.
+    * @return The starting Action, the passed in list, and then a done Action.
+    */
+  def annotateTimestampedActions(description: String, loggingLabelValue: String, isAlwaysRun: Boolean = false)
+                                (actions: List[Action]): List[Action] = {
+    val flags = if (isAlwaysRun) List(ActionFlag.AlwaysRun) else List()
+    val labels = Map(Key.Logging -> loggingLabelValue)
+    val starting = logTimestampedAction(s"Starting $description.", flags, labels)
+    val done = logTimestampedAction(s"Done $description.", flags, labels)
+    starting +: actions :+ done
```

Comment:
This is slightly confusing as to how you're building the list here.  Consider using `::` or `++` ?

Reply:
I suppose it's arguable, i.e. fine as is.  I just had to stare at the different operators for more than a second to figure out what was going on .

Reply:
Heh, it wouldn't be scala without confusing operators!

Will change everything to `List()` and the operators to `++`.